go/libraries/doltcore/remotestorage/map_chunk_cache.go

```diff
@@ -17,132 +17,72 @@ package remotestorage
 import (
 	"sync"
 
+	lru "github.com/hashicorp/golang-lru/v2"
+
 	"github.com/dolthub/dolt/go/store/hash"
 	"github.com/dolthub/dolt/go/store/nbs"
 )
 
-const (
-	// averageChunkSize is used to estimate the size of chunk for purposes of avoiding excessive memory usage
-	averageChunkSize = 1 << 12
-)
-
-// mapChunkCache is a ChunkCache implementation that stores everything in an in memory map.
+// mapChunkCache is a simple ChunkCache implementation that stores
+// cached chunks and has records in two separate lru caches.
 type mapChunkCache struct {
-	mu          *sync.Mutex
-	hashToChunk map[hash.Hash]nbs.ToChunker
-	toFlush     map[hash.Hash]nbs.ToChunker
-	cm          CapacityMonitor
+	mu     *sync.Mutex
+	chunks *lru.TwoQueueCache[hash.Hash, nbs.ToChunker]
+	has    *lru.TwoQueueCache[hash.Hash, struct{}]
 }
 
+const defaultCacheChunkCapacity = 32 * 1024
+const defaultCacheHasCapacity = 1024 * 1024
+
 func newMapChunkCache() *mapChunkCache {
-	return &mapChunkCache{
-		&sync.Mutex{},
-		make(map[hash.Hash]nbs.ToChunker),
-		make(map[hash.Hash]nbs.ToChunker),
-		NewUncappedCapacityMonitor(),
-	}
+	return NewMapChunkCacheWithCapacity(defaultCacheChunkCapacity, defaultCacheHasCapacity)
 }
 
-// used by DoltHub API
-func NewMapChunkCacheWithMaxCapacity(maxCapacity int64) *mapChunkCache {
+func NewMapChunkCacheWithCapacity(maxChunkCapacity, maxHasCapacity int) *mapChunkCache {
+	chunks, err := lru.New2Q[hash.Hash, nbs.ToChunker](maxChunkCapacity)
+	if err != nil {
+		panic(err)
+	}
+	has, err := lru.New2Q[hash.Hash, struct{}](maxHasCapacity)
+	if err != nil {
+		panic(err)
+	}
 	return &mapChunkCache{
 		&sync.Mutex{},
```

Comment:
we don't seem to be using the mutex in any of the methods below, intentional?

Reply:
Good catch! It's no longer necessary...